main.tf

```diff
@@ -0,0 +1,148 @@
+/**
+ * Creates a lambda function with associated role and policies, which
+ * will log to Cloudwatch Logs.
+ *
+ * Creates the following resources:
+ *
+ * * Lambda function
+ * * IAM role and policies described in the inputs
+ * * Cloudwatch Logs group
+ *
+ * ## Usage
+ *
+ * ```hcl
+ * module "my_lambda_function" {
+ *   source                = "trussworks/terraform-aws-lambda"
+ *   name                  = "my_app"
+ *   job_identifier        = "instance_alpha"
+ *   runtime               = "go1.x"
+ *   lambda_role_policy    = "${data.aws_iam_policy_document.my_app_policy.json}"
+ *   s3_bucket             = "my_s3_bucket"
+ *   s3_key                = "my_app/1.0/my_app.zip"
+ *
+ *   subnet_ids            = ["subnet-0123456789abcdef0"]
+ *   security_group_ids    = ["sg-0123456789abcdef0"]
+ *
+ *   source_types          = ["events"]
+ *   source_arns           = ["${aws_cloudwatch_event_rule.trigger.arn}"]
+ *
+ *   env_vars {
+ *     VARNAME = "value"
+ *   }
+ *
+ *   tags {
+ *     "Name" = "my_app_lambda"
```

Comment:
Maybe not the best tag example? I'd expect the `Name` tag to be set automatically. Is that not the case?

Reply:
I don't think any tags get created automatically -- there's a difference between `function_name` and the *tag* Name, the same way there's a difference between the AMI `name` field and the tag Name on AMIs. Or are you saying we should automatically add that tag to every lambda? I'm wary of having too many automatically created things.

Reply:
Hmm, so it sounds like for Lambda's the "Name" tag is arbitrary and not really used? If that's the case, then I still think it's not a good tag example. In AWS, the "Name" tag often is special, hence my questions.